### DISCUSSION TITLE
V4版本配置文件一直跑不起来，哪位大神帮看看到底哪里出问题

### DISCUSSION THREAD

**Discussion by clash1step** [IrineSistiana/mosdns#446] [Q&A]
log:
  level: error
  file: ./mosdns.log

data_providers:
  - tag: geosite
    file: ./geosite.dat
    auto_reload: true
  - tag: geoip
    file: ./geoip.dat
    auto_reload: true

plugins:

  # 转发至本地服务器的插件
  - tag: forward_local
    type: fast_forward
    args:
      upstream:
        - addr: tcp://223.6.6.6
          trusted: true # 是否是可信服务器

  # 转发至远程服务器的插件
  - tag: forward_remote
    type: fast_forward
    args:
      upstream:
        - addr: 127.0.0.1:5352
          trusted: true # 是否是可信服务器

  # 匹配本地域名的插件
  - tag: query_is_local_domain
    type: query_matcher
    args:
      domain:
        - 'provider:geosite:cn'
        - 'provider:geosite:apple-cn'
        - 'provider:geosite:google-cn'

  # 匹配非本地域名的插件
  - tag: query_is_non_local_domain
    type: query_matcher
    args:
      domain:
        - 'provider:geosite:geolocation-!cn'

  # 匹配本地 IP 的插件
  - tag: response_has_local_ip
    type: response_matcher
    args:
      client_ip:
        - 'provider:geoip:cn'

  # 匹配广告域名的插件
  - tag: query_is_ad_domain
    type: query_matcher
    args:
      domain:
        - 'provider:geosite:category-ads-all'

  # 主要的运行逻辑插件
  # sequence 插件中调用的插件 tag 必须在 sequence 前定义，
  # 否则 sequence 找不到对应插件。
  - tag: main_sequence
    type: sequence
    args:
      exec:
        - _default_cache
        # 屏蔽广告域名
        - if: query_is_ad_domain
          exec:
            - _new_nxdomain_response
            - _return

        # 已知的本地域名用本地服务器解析
        - if: query_is_local_domain
          exec:
            - forward_local
            - _return

        # 已知的非本地域名用远程服务器解析
        - if: query_is_non_local_domain
          exec:
            - forward_remote
            - _return

          # 剩下的未知域名用 IP 分流。
          # 这里借助了 `fallback` 工作机制。分流原理请参考 `fallback`
          # 的工作流程。
          # primary 从本地服务器获取应答，丢弃非本地 IP 的结果。
        - primary:
            - forward_local
            - if: "! response_has_local_ip"
              exec:
                - _drop_response
          secondary:
            - forward_remote
          # 这里建议设置成 local 服务器正常延时的 2~5 倍。
          # 这个延时保证了 local 延时偶尔变高时，其结果不会被 remote 抢答。
          # 如果 local 超过这个延时还没响应，可以假设 local 出现了问题。
          # 这时用就采用 remote 的应答。单位: 毫秒。
          fast_fallback: 500
          always_standby: true

servers:
  - exec: main_sequence
    listeners:
      - protocol: udp
        addr: 0.0.0.0:53
      - protocol: tcp
        addr: 0.0.0.0:53

**Comment by kulongwangzhi85** [comment]
你可以用命令方式，在前台运行，看看报什么错
mosdns start -c /etc/mosdns/config.yaml -d /etc/mosdns/

**Comment by hoicau** [comment]
v4版本有一个很神奇的问题，config文件名中不能含有`-`这个符号，否则就开不起来，提示找不到文件，必须用`_`替代
